newIDE/app/src/Profile/RedeemCodeDialog/index.js

```diff
@@ -0,0 +1,123 @@
+// @flow
+import { t, Trans } from '@lingui/macro';
+
+import React from 'react';
+import FlatButton from '../../UI/FlatButton';
+import Dialog, { DialogPrimaryButton } from '../../UI/Dialog';
+import { type AuthenticatedUser } from '../AuthenticatedUserContext';
+import { ColumnStackLayout } from '../../UI/Layout';
+import SemiControlledTextField from '../../UI/SemiControlledTextField';
+import LeftLoader from '../../UI/LeftLoader';
+import { redeemCode } from '../../Utils/GDevelopServices/Usage';
+import { extractGDevelopApiErrorStatusAndCode } from '../../Utils/GDevelopServices/Errors';
+
+type Props = {|
+  onClose: (hasJustRedeemedCode: boolean) => Promise<void>,
+  authenticatedUser: AuthenticatedUser,
+|};
+
+export const getRedeemCodeErrorText = (error: ?Error) => {
+  if (!error) return undefined;
+
+  const extractedStatusAndCode = extractGDevelopApiErrorStatusAndCode(error);
+  if (extractedStatusAndCode) {
+    if (extractedStatusAndCode.status === 404) {
+      return (
+        <Trans>
+          This code is not valid - verify you've entered it properly.
+        </Trans>
+      );
+    }
+    if (
+      extractedStatusAndCode.code ===
+      'redemption-code/cannot-be-redeemed-anymore'
+    )
+      return <Trans>This code was valid but can't be redeemed anymore.</Trans>;
```

Comment:
Should we add some reassurance in case the user is not that one that redeemed the code in the first place? I'm having trouble finding a good phrasing but something like "Contact us if this is unexpected"

Reply:
I gave a try at a sentence to explain this